core/arch/arm/kernel/thread_a32.S

```diff
@@ -741,6 +742,17 @@ END_FUNC thread_unwind_user_mode
 	cps	#CPSR_MODE_SVC
 	mov	sp, r0
 
+#ifdef CFG_CORE_WORKAROUND_NSITR_CACHE_PRIME
+	/*
+	 * Prevent leaking information about which entries has been used in
+	 * cache. We're relying on the secure monitor/dispatcher to take
+	 * care of the BTB.
```

Comment:
Looks like we need also to add a `write_bpiall` instruction at [`.sm_ret_to_nsec`](https://github.com/OP-TEE/optee_os/blob/3.6.0/core/arch/arm/sm/sm_a32.S#L143), no?

Reply:
That already taken care of at:
https://github.com/OP-TEE/optee_os/blob/fcd00ceac7b4b754c40b506fdb0ff204451eb007/core/arch/arm/sm/sm_a32.S#L281-L292

Reply:
This applies to cortex-a8/a9/a17 (and cortex-a15) and are conditioned by `CFG_CORE_WORKAROUND_SPECTRE_BP=y`. 
Other CPUs fallback to standard `sm_vect_table`, without specific processing.

From my understanding `CFG_CORE_WORKAROUND_NSITR_CACHE_PRIME` target secure to non-secure transition only,  right?

Reply:
You're right. I'll try to come up with something.

Reply:
thanks